src/components/sidenav/sidenav.js

```diff
@@ -257,9 +266,40 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
     scope.$watch('isOpen', updateIsOpen);
 
 
+
     // Publish special accessor for the Controller instance
     sidenavCtrl.$toggleOpen = toggleOpen;
 
+    function revalidateVisibility() {
+      var lastValue = scope.isOpen;
+      if (attr.mdIsLockedOpen) return;
+
+      if (isHidden(true, true) && scope.isOpen) {
+        scope.isOpen = false;
+      } else if (!isHidden(true, true) && !scope.isOpen) {
+        scope.isOpen = true;
+      }
+
+      // If the revalidated isOpen Value got changed,
+      // we should apply it to the view without running the updateOpen watcher
+      if (lastValue != scope.isOpen) {
+        skipSidenav = true;
+        if (!scope.$$phase) scope.$apply();
+        skipSidenav = false;
+      }
+    }
+
+    /**
+     * Checks the sidenav's hide state
+     * @param computedStyle Check Computed Style
+     * @param offsetParent Check Offset Parent
+     */
+    function isHidden(computedStyle, offsetParent) {
+      if (computedStyle && window.getComputedStyle(element[0]).display === 'none') return true;
+      if (offsetParent && element[0].offsetParent == null) return true;
```

Comment:
Yes that's also possible, but I thought we should have a clear variant, because this method is really important and needs to be easy to understand, but if you want I can simplify the code?